src/labeled_pc_map_node.cpp

```diff
@@ -14,13 +14,14 @@
 #include <iostream>
 
 // for nclt: 14 classes
-POINT_CLOUD_REGISTER_POINT_STRUCT(pcl::PointSegmentedDistribution<14>,
+POINT_CLOUD_REGISTER_POINT_STRUCT(pcl::PointSegmentedDistribution<21>,
```

Comment:
you changed the number of classes from 14 to 21, does it mean that this code cannot work with the old pretrained network that only have 14 classes

Reply:
Yes. I thought it would make sense to change the hard-coded number to the variable storing the number of classes, but forgot to implement it. The authors hard-coded a lot of numbers, and it's a bit annoying.

Reply:
I find that a little bit annoying that we cannot reused the old weights ( without changing this number back to 14 I mean )

Reply:
It's possible, but I'll have to make changes to their code, to remove these numbers, and it might take some time

Reply:
How would you do that ?

Reply:
I would try to do something like passing parameters from the launch file to the .cpp files, and replacing hard-coded numbers with variables. Something like this:
`int NUM_OF_CLASSES;
NUM_OF_CLASSES = {somehow reading from launch file...};
POINT_CLOUD_REGISTER_POINT_STRUCT(pcl::PointSegmentedDistribution<NUM_OF_CLASSES>`

Reply:
That won't work. The value of a template argument has to be known at compile time, but with what you are planning to do it's only known at runtime.

Reply:
I see. Then I don't fully understand why they made the class templated on the number of classes, and put that parameter in the launchfile. I suppose what I said would require to change the code a little, I don't know if it's that necessary...